### PR TITLE
test(craft): move unit tests to discovered paths and prune trivial tests

### PR DESCRIPTION
**Pull request by wenxi-onyx**
## Description

**Test relocation (discovery fix).** Craft unit tests lived in two locations CI didn't reliably pick up (`backend/tests/unit/build/` and `backend/tests/unit/onyx/server/features/build/`). They now live under `backend/tests/unit/onyx/server/features/craft/`, matching the standard unit-test discovery paths. The Helm-chart-rendering pod-spec test moved to `backend/tests/external_dependency_unit/craft_helm/` with a `setup-helm` step added to the external-dependency-unit workflow (it needs the `helm` binary). The craft-compose workflow's file paths and the craft docs references are updated accordingly.

**Suite cleanup.** Audited all ~380 moved tests for tautological/trivial coverage:
- Removed 9 zero-value tests: a k8s sidecar-host test that asserted its own mock resolver, enum/getter/Pydantic-`extra="forbid"` framework tests, an assert-absence-of-"UTC" test, a duplicate pass-through test, two event-bus scoping tests fully subsumed by `test_event_bus.py`, and a grep-the-source-for-an-import test.
- Fixed 3 tests in `test_docker_manager_config.py` that asserted against `SANDBOX_EXEC_USER`/`SANDBOX_EXEC_ENV`/`SANDBOX_TMPFS_OPTIONS` imported from the module under test; they now pin the hardcoded spec values (`1000:1000`, sandbox HOME/USER env, `rw,nosuid,nodev,size=5g,mode=1777`), so a regression of the constants actually fails.

**New coverage for the real intent gaps left by the deletions:**
- `docker_e2e/test_webapp_preview_docker.py`: docker counterpart of the k8s webapp preview test — pins that `DockerSandboxManager` launches the shared Next.js start script and the preview serves at the basePath through the proxy (previously only "tested" by grepping source text, and only covered end-to-end on k8s).
- `test_parallel_sessions_on_shared_sandbox_stream_independently` in `docker_e2e/test_serve_streaming_docker.py`: two sessions for one user on the same sandbox, streaming concurrent turns — each SSE stream must deliver its own session's text and terminator (the May 2026 unscoped-`/event`-bus hang, previously untestable at unit level).

Supporting refactors: shared `webapp_preview.py` helpers used by both k8s and docker preview suites, `tests/common/paths.py` marker-based repo-root resolution (replaces `parents[N]` indexing that breaks on file moves), `remove_container`/`live_openai_provider` fixture extraction, and a `headless` passthrough on the docker `provision_sandbox` fixture.

## How Has This Been Tested?

- Full moved unit suite locally: `uv run pytest backend/tests/unit/onyx/server/features/craft` — 659 passed.
- `pre-commit` clean across the change set (ty, ruff, workflow lint).
- Integration tests (docker_e2e webapp preview + parallel sessions, craft_helm shard) run in their CI lanes; the compose integration workflow auto-discovers the new `docker_e2e` test file as a shard.

## Additional Options

- [ ] [Optional] Please cherry-pick this PR to the latest release version.
- [x] [Optional] Override Linear Check

---
## Summary by cubic
Move Craft unit tests to standard discovery paths, prune trivial cases, and harden Docker config assertions. Add Docker webapp preview and parallel streaming coverage; CI now vendors `helm` chart deps and fails loudly when missing.

- **Refactors**
  - Moved unit tests to `backend/tests/unit/onyx/server/features/craft/`; moved Helm pod-spec tests to `backend/tests/external_dependency_unit/craft_helm/`; updated workflow paths and docs.
  - Replaced parent-index pathing with `tests.common.paths.find_ancestor_containing`; shared webapp preview helpers for k8s/docker; extracted `remove_container` and `live_openai_provider`; added `headless` to Docker `provision_sandbox`.
  - Pruned trivial tests; pinned Docker spec assertions (`1000:1000`, HOME/USER env, `/tmp` tmpfs). CI installs `helm` via `azure/setup-helm` (`v3.19.0`) and runs `helm dependency build`; `craft_helm` shard fails in CI and skips locally.

- **New Tests**
  - Docker webapp preview: verifies `DockerSandboxManager` runs the shared Next.js start script and the basePath serves through the proxy.
  - Parallel sessions on one sandbox: attach SSE collectors immediately and assert each stream contains only its session’s chunks and terminator (fixes attach timing to avoid 409s).

<sup>Written for commit 005ee52859a1a86b6ecf21166b84b222dcb72128. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/onyx-dot-app/onyx/pull/13523?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: .github/workflows/pr-external-dependency-unit-tests.yml [hunk 1]
```diff
@@ -116,6 +117,12 @@ jobs:
       - name: Setup Playwright
         uses: ./.github/actions/setup-playwright
 
+      - name: Setup Helm
+        if: matrix.test-dir == 'craft_helm'
+        uses: azure/setup-helm@dda3372f752e03dde6b3237bc9431cdc2f7a02a2 # ratchet:azure/setup-helm@v5.0.0
```
[the remark this answers is not in the document]
Comment (wenxi-onyx) [reply .github/workflows/pr-external-dependency-unit-tests.yml:122]: Valid — `charts/*.tgz` is gitignored, so a clean checkout couldn't render and every test skipped green. Fixed in 2f2f1229d7: the shard now registers each repo pinned in `Chart.lock` and runs `helm dependency build` before pytest, and the test's skip-on-render-failure now `pytest.fail`s when `CI` is set so an empty render can never report green. Verified against a `git archive` clean copy with an isolated helm config: all 8 deps vendor and `helm template --show-only templates/sandbox-podtemplate.yaml` renders.

File: .github/workflows/pr-external-dependency-unit-tests.yml [hunk 2]
```diff
@@ -116,6 +117,12 @@ jobs:
...
+        with:
+          version: v3.19.0
```
[the remark this answers is not in the document]
Comment (wenxi-onyx) [reply .github/workflows/pr-external-dependency-unit-tests.yml:124]: Fixed in 2f2f1229d7 — added a "Build Helm chart dependencies" step to the `craft_helm` shard that derives repo URLs from `Chart.lock` (no hardcoded repo list to go stale) and runs `helm dependency build`. Also hardened the test helper: render failures/missing helm now fail instead of skip when `CI` is set.

**Comment by wenxi-onyx** [comment]
@greptile @codex re-review

**Review by Danelegend — APPROVED** [review]
